### DISCUSSION TITLE
Converting an "amount" field in a model from string to float

### DISCUSSION THREAD

**Discussion by prabhur** [pydantic/pydantic#8747] [Question] [answered]
I'm trying to read data from a Google Sheet and trying to create a Model object (pydantic 2.5). 
The "amount" field in the Google Sheet has values with commas (1,000). In the model, this field is declared as a "float". So there is a validation error.

Which function can I use to get rid of the commas? Should I do this before assigning it to the Model or should I use computed_field or validator to perform such pre-processing? Thanks.

**Comment by uriyyo** [comment, marked as answer]
Hi @prabhur,

You can define custom data type, it can look like this:
```py
from typing import Annotated

from pydantic import BeforeValidator


def _prepare_comma_seperated_float(value: str | float) -> float | str:
    if isinstance(value, str):
        return value.replace(",", ".")

    return value


CommaSeperatedFloat = Annotated[
    float,
    BeforeValidator(_prepare_comma_seperated_float),
]
```

Usage:
```py
from pydantic import BaseModel

class Model(BaseModel):
    value: CommaSeperatedFloat


m = Model(value="1,2")
print(m)
```
```py
value=1.2
```

**Reply by prabhur** [reply]
Thanks @uriyyo - this is more natural way of representing the "amount" as a type.
